### PR TITLE
Profiler: measure execution time from the beginning.

### LINKED ISSUE #32753: Profiler shows incorrect execution time
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When using the Profiler, I notice that the execution time is wrong.
In fact the start_time is when `Profiler::__construct` is called https://github.com/PrestaShop/PrestaShop/blob/ff3780e47ad5ff031440c6856f6ec031801de0d3/tools/profiling/Profiler.php#L76,
but it should be the start of the request, when `index.php` is starting to execute.

Immagine 2023-05-30 153824

In this image, the three numbers displayed at the top are `{$smarty.server.REQUEST_TIME_FLOAT} {$run.startTime}
  {load_time data=($run.startTime - $smarty.server.REQUEST_TIME_FLOAT)}` added in https://github.com/PrestaShop/PrestaShop/blob/develop/tools/profiling/templates/run.tpl

This proves that there are 91ms that are used by the initialisation of Prestashop before `Profiler::__construct()`

### Expected behavior

In the profiler, the time should start to run when the script starts executing.
This would be the good starting time on which the calculations for the profiler should be base on: https://github.com/PrestaShop/PrestaShop/blob/ff3780e47ad5ff031440c6856f6ec031801de0d3/config/config.inc.php#L38
If the server exposes the variable `$_SERVER['REQUEST_TIME_FLOAT']`, then this would also be a good candidate as start time.

The correct value for `config` Time in the image above should be 5.611ms + 91.627ms = 97.238ms

### Steps to reproduce

1. Enable profiling in `defines_custom.inc.php` by adding `define('_PS_DEBUG_PROFILING_', true);`
2. In https://github.com/PrestaShop/PrestaShop/blob/ff3780e47ad5ff031440c6856f6ec031801de0d3/tools/profiling/templates/run.tpl#L26, before the `<table>` add `{$smarty.server.REQUEST_TIME_FLOAT} {$run.startTime}
  {load_time data=($run.startTime - $smarty.server.REQUEST_TIME_FLOAT)}`
3. Load a page in the front end and look at the output of the Profiler

### PrestaShop version(s) where the bug happened

8.0.4 and devel

### PHP version(s) where the bug happened

any

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

Société Biblique de Genève

### PR DESCRIPTION
**Pull request by lmeyer1**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Currently the Profiler measures execution time since the call to `Profiler::__cunstruct()`. We need a beter estimate. If the server exposes the variable `$_SERVER['REQUEST_TIME_FLOAT']`, use it, else use the `$start_time` from https://github.com/PrestaShop/PrestaShop/blob/2aaa49f97f372a0efe8e91bdba045ff3617ee5c2/config/config.inc.php#L38
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Look at #32753 to see the changes
| Fixed ticket?     | Fixes #32753
| Related PRs       | none
| Sponsor company   | Société Bilique de Genève

### CODE REVIEW

**Review by matthieu-rolland — APPROVED** [review]
Nice improvement :+1: 

I didn't know about `REQUEST_TIME_FLOAT`

**Review by boherm — APPROVED** [review]
LGTM, thanks @lmeyer1! :)

**Comment by kpodemski** [comment]
I think it could be tested by a dev. WDYT @PrestaShop/qa-functional?

**Review by M0rgan01 — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @lmeyer1 

Thanks for your PR 🚀 
We need to wait the maintainers approval on the related issue to test the PR.

Thanks for your patience 🙏

**Comment by kpodemski** [comment]
@hibatallahAouadni I think we could agree that maintainers reviewing PR is equal to the issue being accepted.

**Review by hibatallahAouadni — APPROVED** [review]
Hello @lmeyer1 

LGTM, QA ✅ 
image

Thanks!

**Comment by nicosomb** [comment]
Thank you @lmeyer1 !

**Comment by lmeyer1** [comment]
Thank you @nicosomb @hibatallahAouadni @kpodemski

**Pull request merged**
